.buildkite/commands/run-ui-tests.sh

```diff
@@ -39,6 +39,9 @@ fi
 echo "--- 📦 Zipping test results"
 cd build/results/ && zip -rq JetpackUITests.xcresult.zip JetpackUITests.xcresult && cd -
 
+echo "--- 💥 Collecting Crash reports"
+for f in ~/Library/Logs/DiagnosticReports/*; do if [[ "$f" == *.ips* ]]; then cp "$f" "build/results/[CRASH]$(basename "$f")"; fi; done
```

Comment:
I'd like to have a simpler command here but `for` + `if` was needed so it could work for both cases when there is and there is not `.ips` file. `for f in ~/Library/Logs/DiagnosticReports/*.ips` exits with non-zero status when no `.ips` files are present.

Reply:
got a noob question, are all `.ips` files crashes? 

and small feedback, as a one-liner, i find it a bit hard to read, what do you think about breaking it into multiple lines?

```suggestion
for f in ~/Library/Logs/DiagnosticReports/*; do
  if [[ "$f" == *.ips* ]]; then
    cp "$f" "build/results/[CRASH]$(basename "$f")"
  fi
done
```

also i think might be good to get some 👀  from @wordpress-mobile/apps-infrastructure for this one

Reply:
> I'd like to have a simpler command here but `for` + `if` was needed so it could work for both cases when there is and there is not `.ips` file. `for f in ~/Library/Logs/DiagnosticReports/*.ips` exits with non-zero status when no `.ips` files are present.

I think you can circumvent this limitation by setting `shopt -s nullglob` (at least according to [this bash cheatsheet](https://devhints.io/bash#glob-options)).

_From [what I read in StackExchange](https://unix.stackexchange.com/a/204944), if we were using `zsh` instead of `bash` we'd have the ability to apply the nullglob option locally just for a specific glob pattern with the `(N)`. But I think it's better to stick with `bash` and use `shopt -s nullglob` there. In case we want to make sure this is only applied locally, maybe we could either reset the option to its default once done, or use a subshell like `$(shopt -s nullglob; echo *.ips)` though I'm not sure that's better._

Another alternative is to use `find`, as in `find ~/Library/Logs/DiagnosticReports/ -name '*.ips' -exec cp {} build/results/ \;` or something like that (would not prepend the `[CRASH]` string to the basename though). Or still using the `for` loop to have more control but using find for the listing:
```
for f in $(find ~/Library/Logs/DiagnosticReports -name '*.ips'); do
  mv "$f" "build/results/[CRASH]$(basename "$f")"
done
```

> got a noob question, are all .ips files crashes?

That's a very good question, which I don't have the answer to. TBH, TIL about `.ips` files in the first place 🙃

Reply:
> got a noob question, are all .ips files crashes?

Good question, @jostnes! Not all `.ips` files are crashes, and I didn't know that 😅 . ChatGPT says `The .ips file extension is used by Xcode for a variety of purposes, including instrumenting iOS apps for debugging, profiling, and performance analysis.`, but the file location for each purpose is different. Profiling and performance analysis, as an example, are found in `~/Library/Developer/Xcode/DerivedData`. The subdirectory we're looking into, `~/Library/Logs/DiagnosticReports`, would hold `.ips` files for debugging and crash reports, which sounds ok for our use in CI.

> and small feedback, as a one-liner, i find it a bit hard to read, what do you think about breaking it into multiple lines?

Yep, it was a bit too much. It's a bit better now with [`cfa2449`](https://github.com/wordpress-mobile/WordPress-iOS/pull/20649/commits/cfa2449e6e856500e45379489974b71bc27759b9).

> I think you can circumvent this limitation by setting `shopt -s nullglob`

Thanks for all the alternatives, @AliSoftware! I tried it locally in a subshell and although it worked fine, I decided to go with the whitespaces/newline proof option mainly because while the current filename format is ok, Apple can change next week. Changes in [`cfa2449`](https://github.com/wordpress-mobile/WordPress-iOS/pull/20649/commits/cfa2449e6e856500e45379489974b71bc27759b9).